File: rules/opengrep-rules/go/security.test.go
package security_test

import "os/exec"

// Annotated tests for go/security.yaml.
func tainted(arg string) {
	// ruleid: scp.go.injection.command
	exec.Command("sh", "-c", arg)
}

func safe(arg string) {
	// ok: scp.go.injection.command
	exec.Command("echo", arg)
}

File: rules/opengrep-rules/java/security.test.java
import java.security.MessageDigest;

public class SecurityTest {
    // Annotated tests for java/security.yaml.
    public void tainted(String input) throws Exception {
        // ruleid: scp.java.injection.command
        Runtime.getRuntime().exec(input);
        // ruleid: scp.java.crypto.weak-md5
        MessageDigest.getInstance("MD5");
    }

    public void safe(String input) throws Exception {
        // ok: scp.java.crypto.weak-md5
        MessageDigest.getInstance("SHA-256");
    }
}
